src/jacobians.jl

```diff
@@ -1,4 +1,4 @@
-struct JacobianCache{CacheType1,CacheType2,CacheType3,ColorType,fdtype,returntype,inplace}
+mutable struct JacobianCache{CacheType1,CacheType2,CacheType3,ColorType,fdtype,returntype,inplace}
```

Comment:
this shouldn't be necessary?

Reply:
This is necessary because we change `color`'s value

Reply:
oh yes.